typescript/sdk/src/timelock/evm/EvmTimelockDeployer.hardhat-test.ts

```diff
@@ -164,4 +171,187 @@ describe('EvmTimelockDeployer', async () => {
+    );
+    const timelockAddress = TimelockController.address;
+
+    const timelock = TimelockController__factory.connect(
+      timelockAddress,
+      multiProvider.getProvider(TestChainName.test4),
+    );
+
+    expect(
+      await timelock.hasRole(EXECUTOR_ROLE, hre.ethers.constants.AddressZero),
+    ).to.be.true;
+
+    const _0_BYTES_32 =
+      '0x0000000000000000000000000000000000000000000000000000000000000000';
+    const updatedDelay = 100;
+    const testTxData =
+      TimelockController__factory.createInterface().encodeFunctionData(
+        'updateDelay',
+        [updatedDelay],
+      );
+
+    const signerTimelock = TimelockController__factory.connect(
+      timelockAddress,
+      signer,
+    );
+    const tx = await signerTimelock.schedule(
+      timelockAddress,
+      0,
+      testTxData,
+      _0_BYTES_32,
+      _0_BYTES_32,
+      0,
+    );
+    await tx.wait();
+
+    const randomSignerTimelockInstance = TimelockController__factory.connect(
+      timelockAddress,
+      otherSigner,
+    );
+    const executeTx = await randomSignerTimelockInstance.execute(
+      timelockAddress,
+      0,
+      testTxData,
+      _0_BYTES_32,
+      _0_BYTES_32,
+    );
+    await executeTx.wait();
+
+    expect(await timelock.getMinDelay()).to.equal(updatedDelay);
+  });
+
+  async function assertCancellerConfig(
+    expectedConfig: TimelockConfig,
+    timelockInstance: TimelockController,
+  ) {
+    // proposer should be a proposer but not a canceller if not in canceller config
+    assert(expectedConfig.proposers, 'Expected proposers to be defined');
+    for (const proposer of expectedConfig.proposers) {
+      expect(await timelockInstance.hasRole(PROPOSER_ROLE, proposer)).to.be
+        .true;
+
+      if (!expectedConfig.cancellers?.includes(proposer)) {
+        expect(await timelockInstance.hasRole(CANCELLER_ROLE, proposer)).to.be
+          .false;
+      }
+    }
+
+    // cancellers should be a canceller but not a proposer if not in the proposer config
+    assert(expectedConfig.cancellers, 'Expected cancellers to be defined');
+    for (const canceller of expectedConfig.cancellers) {
+      expect(await timelockInstance.hasRole(CANCELLER_ROLE, canceller)).to.be
+        .true;
+      if (!expectedConfig.proposers?.includes(canceller)) {
+        expect(await timelockInstance.hasRole(PROPOSER_ROLE, canceller)).to.be
+          .false;
+      }
+    }
+
+    // signer should not be the timelock admin after deployment
+    const timelockAdminRoleHash = await timelockInstance.TIMELOCK_ADMIN_ROLE();
+    expect(
+      await timelockInstance.hasRole(timelockAdminRoleHash, signer.address),
+    ).to.be.false;
+
+    // Timelock should still be the admin of itself
+    expect(
+      await timelockInstance.hasRole(
+        timelockAdminRoleHash,
+        timelockInstance.address,
+      ),
+    ).to.be.true;
+
+    // if an admin was set it should be the admin after the changes
+    if (expectedConfig.admin) {
```

Comment:
This seems unrelated to the function name/purpose of asserting canceller role assignments